### PR TITLE
Simultaneous support for Badger V1+V2 and ...

### LINKED ISSUE #189: order IDs changed since last read
### Subject of the issue
smalstep CA server responds with
````
order IDs changed since last read
````

### Your environment
* OS: WSL (Bash on Windows), (Ubuntu)
* Version: WSL 1 (stable), (Windows 10 x64 Pro)

### Steps to reproduce
Up a smallstep CA container and a traefik container that uses the CA container as certresolver.

### Expected behaviour
No such errors, certs are signed correctly.

### Actual behaviour
smallstep CA serverside error `order IDs changed since last read` and the site cert is not signed for traefik.

### Additional context
badger2 database used, persisted in mounted docker volume on host.

### LINKED ISSUE #211: Error running a new CA on linux/arm
### Subject of the issue
After going through the usual CA bootstrap process using step command, I'm not able to start step-ca using the default generated config file.

### Your environment
* OS - Raspbian GNU/Linux 10 (buster) - on Raspberry Pi 4 (4Gb RAM)
* Kernel Version  4.19.97-v7+ #1294 SMP

### Steps to reproduce
1. create a new CA using: `step ca init`
2. run the new CA using:  `step-ca /root/.step/config/ca.json --resolver=8.8.8.8:53`

### Expected behaviour
The new CA should be up & ready to serve connections.

### Actual behaviour
Error message on startup:
`Error opening database of Type badger with source /root/.step/db: error opening Badger database: Unable to mmap RDWR log file: cannot allocate memory`

### Additional context
I've tried rebuilding step-ca (after merging in master the changes from bagder2 branch), and the build fails:
`...\go\pkg\mod\github.com\dgraph-io\badger\v2@v2.0.1\value.go:859:38: constant 4294967295 overflows int`

used build command: 
`env GOOS=linux GOARCH=arm go build -ldflags="-s -w" -o step-ca`

### LINKED ISSUE #215: Expose ValueLogLoadingMode badger option in the configuration.
Required for running `step-ca` on 32bit systems.

https://github.com/smallstep/nosql/issues/6

### LINKED ISSUE #225: Test (and update) to use Badger2 and Badger1 together.
https://github.com/dgraph-io/badger/issues/1208

### PR DESCRIPTION
**Pull request by dopey**
* valueLogLoadingMode config for low RAM badger environments

Fixes #225 
Fixes #211 
Fixes #189 
Fixes #215

### CODE REVIEW

File: db/db.go [hunk 1]
```diff
@@ -34,6 +34,11 @@ type Config struct {
 	DataSource string `json:"dataSource"`
 	ValueDir   string `json:"valueDir,omitempty"`
 	Database   string `json:"database,omitempty"`
+
+	// ValueLogLoadingMode can be set to FileIO (instead of the default
+	// options.MemoryMap) to avoid memory-mapping log files. This can be useful
+	// in environments with low RAM
+	BadgerValueLogLoadingMode string `json:"badgerValueLogLoadingMode"`
```
Comment (maraino) [inline db/db.go:41]: See comment in https://github.com/smallstep/nosql/pull/8
```go
FileLoadingMode string `json:"fileLoadingMode"`
```

File: go.mod [hunk 2]
```diff
@@ -13,7 +13,7 @@ require (
 	github.com/sirupsen/logrus v1.4.2
 	github.com/smallstep/assert v0.0.0-20200103212524-b99dc1097b15
 	github.com/smallstep/cli v0.14.2
-	github.com/smallstep/nosql v0.2.0
+	github.com/smallstep/nosql v0.2.1-0.20200417162056-c1de767c9d0d
```
Comment (maraino) [inline go.mod:16]: Are we using a non-released version?

File: go.mod [hunk 3]
```diff
@@ -24,3 +24,4 @@ require (
 )
 
 //replace github.com/smallstep/cli => ../cli
+replace github.com/smallstep/nosql => ../nosql
```
Comment (maraino) [inline go.mod:27]: Comment this line

File: go.mod (hunk 2 above)
Comment (dopey) [reply go.mod:16]: Yes.

Comment (maraino) [reply go.mod:16]: Just change this after the release :)

**Pull request merged**